plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/Inline.hs

```diff
@@ -273,24 +306,24 @@ maybeAddSubst body a s n rhs = do
             -- See Note [Inlining approach and 'Secrets of the GHC Inliner'] and [Inlining and
             -- purity]. This is the case where we don't know that the number of occurrences is
             -- exactly one, so there's no point checking if the term is immediately evaluated.
-            postUnconditional <- fmap ((&&) termIsPure) (acceptable rhs')
+            let postUnconditional = isTermPure && acceptable rhs'
```

Comment:
Unrelated to this PR, but I think this line is wrong? Even if something occurs multiple times, we don't need to care if it's pure for _non-strict_ bindings, because the effects would have occurred at each use site regardless.

Reply:
Yeah this is unrelated to this PR but I can fix it here I guess. I'll add a `wontDuplicateEffect`, in which I check the strictness first. If it's non-strict, then it's `True` and go on to checking `purity`.

Reply:
Well, the criterion is "is this _binding_ pure, i.e. will evaluating it have effects", so maybe you just want `isTermBindingPure :: Strictness -> Term -> Bool`  or whatever.